### PR TITLE
Enhancement/post

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: pyaedt/application/Analysis.py [hunk 1]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
         """
         return SOLUTIONS()
 
+    @aedt_exception_handler
+    def analyze_all(self):
+        """Analyze all setup in a actual design.
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:413]:
```suggestion
        """Analyze all setup in an actual design.
```

File: pyaedt/application/Analysis.py [hunk 2]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+
+        if not setupname and ":" in self.nominal_sweep:
+            setupname = self.nominal_adaptive.split(":")[0].strip()
+        elif not setupname:
+            self.logger.warning("No Setup defined.")
+            return False
+        if not sweepname and ":" in self.nominal_sweep:
+            sweepname = self.nominal_adaptive.split(":")[1].strip()
+        elif not sweepname:
+            self.logger.warning("No Setup defined.")
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:447]:
```suggestion
            self.logger.warning("No Sweep defined.")
```

File: pyaedt/application/Analysis.py [hunk 3]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+        else:
+            excitations = self.oboundary.GetNumExcitations()
+        reportnames = self.post.oreportsetup.GetAllReportNames()
+        for rName in reportnames:
+            rName2 = rName.replace(" ", "_")
+            self.post.oreportsetup.UpdateReports([str(rName)])
+            szExportPath = os.path.join(export_folder,
+                                        "{0}_{1}_{2}.csv".format(self.project_name, self.design_name, rName2))
+            self.post.oreportsetup.ExportToFile(str(rName), szExportPath)
+            self.logger.debug("Export Data: {}".format(szExportPath))
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:493]: @maxcapodi78 Do we want to log debug or info here?

File: pyaedt/application/Analysis.py [hunk 4]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+        Returns
+        -------
+        bool
+            ``True`` when simulation is finished.
+        """
+        self.odesign.AnalyzeAll()
+        return True
+
+    @aedt_exception_handler
+    def list_of_variations(self, setupname=None, sweepname=None):
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:424]:
```suggestion
    def list_of_variations(self, setup_name=None, sweep_name=None):
```

It is just to respect the snake_case convention.

File: pyaedt/application/Analysis.py [hunk 5]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+            excitations = len(self.oexcitation.GetAllPortsList())
+        else:
+            excitations = self.oboundary.GetNumExcitations()
+        reportnames = self.post.oreportsetup.GetAllReportNames()
+        for rName in reportnames:
+            rName2 = rName.replace(" ", "_")
+            self.post.oreportsetup.UpdateReports([str(rName)])
+            szExportPath = os.path.join(export_folder,
+                                        "{0}_{1}_{2}.csv".format(self.project_name, self.design_name, rName2))
+            self.post.oreportsetup.ExportToFile(str(rName), szExportPath)
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:492]:
```suggestion
        report_names = self.post.oreportsetup.GetAllReportNames()
        for report_name in report_names:
            report_name2 = report_name.replace(" ", "_")
            self.post.oreportsetup.UpdateReports([str(report_name)])
            szExportPath = os.path.join(export_folder,
                                        "{0}_{1}_{2}.csv".format(self.project_name, self.design_name, report_name2))
            self.post.oreportsetup.ExportToFile(str(report_name), szExportPath)
```

File: pyaedt/application/Analysis.py [hunk 6]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+                                                            "{0}_{1}.s{2}p".format(self.project_name, varCount,
+                                                                                   excitations))
+                                self.logger.info("Export SnP: {}".format(szExportPath))
+                                self.osolution.ExportNetworkData(variation, ["{0}:{1}".format(s, sweep)], 3,
+                                                                 szExportPath,
+                                                                 ["All"], True, 50, "S", -1, 0, 15, True, False, False)
+                                exported_files.append(szExportPath)
+                                self.logger.info("Export Touchstone: %s", szExportPath)
+                            except:
+                                self.logger.info("Export SnP Failed: No Solution")
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:551]: Should it be a warning or an error message here if the Snp export fails?

**Review by MaxJPRey — APPROVED** [review]

File: pyaedt/application/Analysis.py (hunk 3 above)
Comment (maxcapodi78) [reply pyaedt/application/Analysis.py:493]: info

File: pyaedt/application/Analysis.py (hunk 6 above)
Comment (maxcapodi78) [reply pyaedt/application/Analysis.py:551]: No, just warning is enough

File: pyaedt/application/Analysis.py [hunk 7]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+                            export_path = os.path.join(export_folder,
+                                                        "{0}.s{1}p".format(self.project_name, excitations))
+                            self.osolution.ExportNetworkData(variation_array[0], ["{0}:{1}".format(s, sweep)], 3,
+                                                             export_path, ["All"], True, 50, "S", -1, 0, 15, True,
+                                                             False,
+                                                             False)
+                            exported_files.append(export_path)
+                            self.logger.info("Export Touchstone: %s", export_path)
+                        except:
+                            self.logger.info("Export SnP Failed: No Solution")
```
Comment (Alberto-DM) [inline pyaedt/application/Analysis.py:524]:
```suggestion
                            self.logger.warning("Export SnP failed: no solutions found")
```

File: pyaedt/application/Analysis.py [hunk 8]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+                                                            "{0}_{1}.s{2}p".format(self.project_name, varCount,
+                                                                                   excitations))
+                                self.logger.info("Export SnP: {}".format(export_path))
+                                self.osolution.ExportNetworkData(variation, ["{0}:{1}".format(s, sweep)], 3,
+                                                                 export_path,
+                                                                 ["All"], True, 50, "S", -1, 0, 15, True, False, False)
+                                exported_files.append(export_path)
+                                self.logger.info("Export Touchstone: %s", export_path)
+                            except:
+                                self.logger.info("Export SnP Failed: No Solution")
```
Comment (Alberto-DM) [inline pyaedt/application/Analysis.py:551]:
```suggestion
                                self.logger.warning("Export SnP failed: no solutions found")
```

File: pyaedt/application/Design.py [hunk 9]
```diff
@@ -1152,6 +1152,7 @@ def export_profile(self, setup_name, variation_string="", file_path=None):
         if not file_path:
             file_path = os.path.join(self.project_path, generate_unique_name("Profile")+".prop")
         self.odesign.ExportProfile(setup_name, variation_string, file_path)
+        self.logger.info("Export Profile to  %s", file_path)
```
Comment (Alberto-DM) [inline pyaedt/application/Design.py:1155]:
```suggestion
        self.logger.info("Exported Profile to file {}".format(file_path))
```

File: pyaedt/application/Analysis.py [hunk 10]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+                            try:
+                                export_path = os.path.join(export_folder,
+                                                            "{0}_{1}.s{2}p".format(self.project_name, varCount,
+                                                                                   excitations))
+                                self.logger.info("Export SnP: {}".format(export_path))
+                                self.osolution.ExportNetworkData(variation, ["{0}:{1}".format(s, sweep)], 3,
+                                                                 export_path,
+                                                                 ["All"], True, 50, "S", -1, 0, 15, True, False, False)
+                                exported_files.append(export_path)
+                                self.logger.info("Export Touchstone: %s", export_path)
```
Comment (Alberto-DM) [inline pyaedt/application/Analysis.py:549]:
```suggestion
                                self.logger.info("Exported Touchstone: %s", export_path)
```

File: pyaedt/application/Analysis.py [hunk 11]
```diff
@@ -408,6 +408,174 @@ def SolutionTypes(self):
...
+                    if self.solution_type in ["HFSS3DLayout", "HFSS 3D Layout Design", "HFSS", "Circuit"]:
+                        try:
+                            export_path = os.path.join(export_folder,
+                                                        "{0}.s{1}p".format(self.project_name, excitations))
+                            self.osolution.ExportNetworkData(variation_array[0], ["{0}:{1}".format(s, sweep)], 3,
+                                                             export_path, ["All"], True, 50, "S", -1, 0, 15, True,
+                                                             False,
+                                                             False)
+                            exported_files.append(export_path)
+                            self.logger.info("Export Touchstone: %s", export_path)
```
Comment (Alberto-DM) [inline pyaedt/application/Analysis.py:522]:
```suggestion
                            self.logger.info("Exported Touchstone: %s", export_path)
```

**Review by Alberto-DM — APPROVED** [review]
Minor changes suggested, otherwise LGTM.
Approved with minor changes.

**Pull request merged**